presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/clientproviders/RetryingPrestoThriftServiceProvider.java

```diff
@@ -68,27 +75,33 @@ public RetryingPrestoThriftServiceProvider(@NonRetrying PrestoThriftServiceProvi
     @Override
     public PrestoThriftService anyHostClient()
     {
-        return new RetryingService(original::anyHostClient, retry);
+        return new RetryingService(() -> (DefaultPrestoThriftService) original.anyHostClient(), retry);
```

Comment:
This code somewhat violates abstractions and creates tight coupling between components. In this case you're not working through the interface, but instead depend on another service giving you an object you need.
Try the following approach.
Extend `PrestoThriftService` interface to add `getLocation()` method to it. Then you can still use your wrappers, but they will be hidden from Default and Retrying providers and they will work though interfaces.